ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/disable_gui/tasks/main.yml

```diff
@@ -0,0 +1,28 @@
+---
+#################################################
+# Ensure that no desktop environment is running #
+#################################################
+
+- name: Check if desktop is installed
+  stat:
+    path: /usr/share/xsessions
+  when: (ansible_distribution == "CentOS" and ansible_distribution_major_version >= "7") or (ansible_distribution == "Ubuntu" and ansible_distribution_major_version >= "18")
+  register: desktop_installed
+  tags: disable_gui
+
+- name: Disable GNOME for CentOS 7+ and Ubuntu 18+
+  shell: systemctl set-default multi-user
+  when:
+    - (ansible_distribution == "CentOS" and ansible_distribution_major_version >= "7") or (ansible_distribution == "Ubuntu" and ansible_distribution_major_version >= "18")
+    - desktop_installed.stat.exists
+  register: gui_disabled
+  tags: disable_gui
+
+- name: Reboot after disabling gui
```

Comment:
Can this not be done using a systemctl command to switch it into console mode instead of a reboot?

Reply:
I use a systemctl command on line 14. From what I have tested, this then needs a reboot to make the changes effective. I have not yet found a single systemctl command which does it without the reboot

Reply:
Have you tried `systemctl isolate multi-user.target`?

Reply:
I have just tried that on a centos8 virtual box vm. Gui immediately turns off, but is left in a 'blinking' console state. I am observing to see if this changes

Reply:
Press ALT-F1, or Alt+F2, etc and see if that shows a login prompt at some point. What you're seeing is almost certainly "correct"

Reply:
alt-f1 brings up a login prompt, but I am struggling to think how to ansibile-ise this

Reply:
There is no need to ansiblise the ALT-F1 bit. Just leave it as is.

Reply:
Ok, I've added a `systemctl isolate multi-user.target` task and removed the reboot one. However, using `isolate` to remove the gui will reset upon a reboot, restarting the gui. So I have kept in the `systemctl set-default multi-user`